patina_dxe_core/src/image.rs

```diff
@@ -2196,4 +2196,317 @@ mod tests {
+        //
+        // Also validates section alignment by directly calling core_load_pe_image().
+        with_locked_state(|| {
+            let mut test_file =
+                File::open(test_collateral!("RustImageTestDxe.efi")).expect("failed to open test file.");
+            let mut image: Vec<u8> = Vec::new();
+            test_file.read_to_end(&mut image).expect("failed to read test file");
+
+            // Test 1: Full load_image flow
+            let mut image_handle: efi::Handle = core::ptr::null_mut();
+            let status = load_image(
+                false.into(),
+                protocol_db::DXE_CORE_HANDLE,
+                core::ptr::null_mut(),
+                image.as_mut_ptr() as *mut c_void,
+                image.len(),
+                core::ptr::addr_of_mut!(image_handle),
+            );
+
+            // With proper GCD setup, this should succeed
+            assert_eq!(status, efi::Status::SUCCESS);
+
+            // Verify the image was loaded successfully with correct properties
+            let private_data = PRIVATE_IMAGE_DATA.lock();
+            let image_data = private_data.private_image_data.get(&image_handle).unwrap();
+            assert_ne!(image_data.entry_point as usize, 0);
+            assert_eq!(image_data.image_info.image_code_type, efi::BOOT_SERVICES_CODE);
+            assert_eq!(image_data.image_info.image_data_type, efi::BOOT_SERVICES_DATA);
+            drop(private_data);
+
+            // Test 2: Direct core_load_pe_image to validate section alignment handling
+            let image_info = empty_image_info();
+            let result = super::core_load_pe_image(&image, image_info);
+
+            // Should load successfully with valid alignment
+            assert!(result.is_ok());
+            let private_info = result.unwrap();
+            assert_ne!(private_info.entry_point as usize, 0);
+
+            // If we get here, memory protections were successfully applied.
+        });
+    }
+
+    #[test]
+    fn apply_memory_protections_should_fail_when_section_address_not_in_gcd() {
+        // This test verifies error path #1: GCD descriptor lookup failure
+        //
+        // Initialize GCD but don't add any memory descriptors, leaving the
+        // internal RBT empty. When get_memory_descriptor_for_address is called,
+        // get_closest_idx returns None (no descriptors in tree), causing NotFound error.
+        //
+        // Normally, the first add_memory_space call creates an initial
+        // NonExistent descriptor covering [0, maximum_address), so all subsequent lookups
+        // succeed. By skipping add_memory_space entirely, we keep the RBT empty, making
+        // ANY address lookup fail with NotFound.
+        //
+        // This scenario represents a corrupted or uninitialized GCD state where memory
+        // descriptors are missing - an edge case that the error handling should catch.
+
+        let result = test_support::with_global_lock(|| unsafe {
```

Comment:
This unsafe block covers more than it needs to, it should be scoped to the actual unsafe operation(s). Please add a `SAFETY:` comment for each new unsafe block as well.